### PR TITLE
Streamline code formatting.

### PR DESCRIPTION
**Pull request by shenlebantongying**
Borrowed from https://github.com/xiaoyifang/goldendict-ng/blob/staged/.clang-format

The style is adjusted to minimize how much code will be changed if run clang-format over the entire code base.

close: https://github.com/goldendict/goldendict/issues/1454

All new codes should follow this.

image

https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 

Fix those things:

image
image
image
image
image
image
image

https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 https://github.com/goldendict/goldendict/pull/1458 

image

### CODE REVIEW

File: .clang-format
Comment (vedgy) [inline .clang-format:78]: This option value definitely does not match the most common GoldenDict style. [PPDIS_None](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#indentppdirectives) is much more common. There are very few preprocessor directive indentations in non-third-party code.

On the other hand, deeply nested preprocessor conditional blocks are difficult to read. Is there a way to ask clang-format to leave preprocessor indentations as is?

File: .clang-format
Comment (vedgy) [inline .clang-format:120]: I think this should be `SBPO_Never` because of https://github.com/goldendict/goldendict/pull/1458#issuecomment-1037393138 (slightly more common) and because recent code changes preferred the `if( condition )` style.

File: .clang-format
Comment (vedgy) [inline .clang-format:7]: Should be `None`, I think.

```bash
$ git grep -n '  =' -- :^winlibs :^maclibs :^MSBuild *.hh *.cc | wc -l
187
``` 

And these matches are mostly limited to a few source files: articleview.cc, mouseover.cc, sapi.hh, uiauto.hh.

File: .clang-format
Comment (vedgy) [inline .clang-format:32]: Should definitely be `Always`. Braces attached after a control statement are rare in GoldenDict codebase.  As it is, there is **a lot** of diff like this:
```diff
-    for( size_t left = baseText.size(); left; )
-    {
-      if( *nextChar >= 0x10000 )
-      {
+    for ( size_t left = baseText.size(); left; ) {
+      if ( *nextChar >= 0x10000 ) {
```

**Review by vedgy — CHANGES_REQUESTED** [review]
Haven't got much time for GoldenDict at the moment. So here is a partial review: most obvious issues in the diff `clang-format -i articleview.cc`.

File: .clang-format
Comment (shenlebantongying) [reply .clang-format:32]: It is stupid, becase the code is using a lot of this style of if

```
if {
}
else {
}
```

The else is not the same line with `if`'s `}`

The old style is self-conflicting.

To ensure good looking of `else` this, the Control Statement must be never.

File: .clang-format
Comment (shenlebantongying) [reply .clang-format:120]: This doesn't matter, this is your personal style.

File: .clang-format
Comment (shenlebantongying) [reply .clang-format]: > Is there a way to ask clang-format to leave preprocessor indentations as is

Go file a bug, your time to shine.

File: .clang-format
Comment (shenlebantongying) [reply .clang-format]: fixed

**Comment by shenlebantongying** [comment]
THIS IS NOT ROCKET SCIENCE.

File: .clang-format
Comment (vedgy) [reply .clang-format:120]: I don't understand your question. By recent code changes, I mean @Abs62's and my commits for the last 5 years. I have used this style to follow the maintainer's preferred style.

File: .clang-format
Comment (vedgy) [reply .clang-format:32]: I have encountered opening brace on the next line much more often than attached to the `if` line in GoldenDict's source code. `try`-`catch` is rare and doesn't matter much. Again, this is the style of almost all new code committed in the past 5 years.

File: .clang-format
Comment (shenlebantongying) [reply .clang-format:120]: You are using double standards: recent code + your one-liner's statics

so that your personal style can be used.

File: .clang-format
Comment (shenlebantongying) [reply .clang-format:32]: > there is a lot of diff like this:

What do u expect?? The code isn't written in a unified style. Of course, there are a lot. It is impossible to change no code nothing and have a working formatter.

Some style are 50%/50%. It doesn't matter, just choose anyone.

Comment (shenlebantongying) [reply .clang-format:32]: > What's the conflict in the existing style

Why don't u try adjust the .clang-format by yourself? This codebase has two styles:

The bracket placements and if else/try catch depend on each other.

```
if (){
}
else {
}
```

and

```
if (){
} else {
}
```

If the change line after `if ()` then you get an insane shit:

```
if ()
{
}
else 
{
}
```

Comment (shenlebantongying) [reply .clang-format:32]: > > The style is adjusted to minimize how much code will be changed if run clang-format over the entire code base.
> In this case your personal preference contradicts the stated principle of the pull request.

Then why are you citing "recent changes"??

File: .clang-format
Comment (shenlebantongying) [reply .clang-format]: > On the other hand, deeply nested preprocessor conditional blocks are difficult to read.

So both are good in some aspect.

Can u make a DECISION as "maintainer"?

File: .clang-format
Comment (vedgy) [reply .clang-format:32]: > If the change line after `if ()` then you get an insane shit:
> 
> ```
> if ()
> {
> }
> else 
> {
> }
> ```

That's exactly the most common style in GoldenDict source code by far. If you reject it as "insane", don't claim objectivity in the PR description and comments.

> Why don't u try adjust the .clang-format by yourself? 

That is certainly preferable to reading and replying to your belligerent and nonconstructive comments. Nothing good comes out of the review at this rate.

Comment (shenlebantongying) [reply .clang-format]: Keep reviewing, this must be "always".

> Nothing good comes out of the review at this rate.

Are u retard or something? You are the reviewer, then review everything at once. GO.

Comment (shenlebantongying) [reply .clang-format]: > Should definitely be Always

> That's exactly the most common style in GoldenDict source code by far

If you see some format stupid, don't say anything because this comes from your review.

Comment (shenlebantongying) [reply .clang-format]: >  your belligerent and nonconstructive comments

You are using triple standards:

* Your one-liner
* "Recent" commits
* Not my style

What the fck do u want???

Comment (shenlebantongying) [reply .clang-format]: image

Comment (vedgy) [reply .clang-format]: With such an attitude and language from the PR author, I won't be reviewing it, at least for a while. Your goal here appears to be insulting reviewers and wasting time anyway.

Comment (shenlebantongying) [reply .clang-format]: Your attitude is the problem.

You are not going to merge anything because this is written in your DNA, in your personality, you cannot lead and cannot make a decision, thus you are afraid to press that merge button. You cannot even merge your own PR.

Without proving other people wrong, you cannot satisfy yourself by reviewing. 

>  I won't be reviewing it

I legitimately don't see any difference. You won't merge it anyway. After weeks of useless work to stratify your stupid reviews and nothing happens https://github.com/goldendict/goldendict/pull/1515

image

🤮 🤮 🤮 🤮 🤮 🤮 🤮  You changed 4k lines in total, and you align yourself with another guy, shameless. 🤮 🤮 🤮 🤮 🤮 🤮 🤮

Comment (shenlebantongying) [reply .clang-format]: Why r u so autistic that you don't even know who can be your friend?? https://github.com/goldendict/goldendict/issues/1424

File: .clang-format
Comment (shenlebantongying) [reply .clang-format]: Make a decision about which one to use??

If you cannot, just stick with my decision. There is no difference.

**Comment by vedgy** [comment]
@shenlebantongying's comments under this pull request reveal his/her true intentions: insult reviewers and generally waste time. For this behavior the pull request's author has been blocked to prevent future abuses.

The value of this pull request's contribution is low: it proposes to add goldendict-ng's _.clang-format_ verbatim, then makes a few fixes requested during code review. Closing it is not a great loss, because the same copying and fixing can be trivially repeated at a later time, without the malicious and obnoxious comments to distract and demotivate.

**Pull request closed without merging**
